readme.md

```diff
@@ -1,7 +1,276 @@
-## An attempt to find an interoperable CSSinJS format.
+## Interoperable CSSinJS format.
 
-The biggest issue at scale we currently have with all the CSSinJS implementations is that they are not sharable without the runtime.
+The biggest issue at scale we currently have with all CSSinJS implementations is that they are not sharable without the runtime.
 
-If we could find a format that is fast an easy to parse and supports all needs of current CSSinJS implementations, it could become a de facto standard for exported CSSinJS.
+We can define a format that is easy to parse and supports all the needs of the current CSSinJS implementations on top of CSS.
 
-It could be similar to how CommonJS and ES3 are currently the standard for sharing JavaScript modules on npm.
+Package maintainers will be able to publish JavaScript files to NPM with this format inside instead of plain CSS.
+
+## First draft
+
+Format is optimized for parsing in JavaScript, not for human readability.
+
+### Markers
+
+Examples are using constant names instead of values for readability within the spec. The resulting format will use their values.
+
+```js
+const RULE_START = 0
+const RULE_END = 1
+const SELECTOR = 2
+const NAME = 3
+const PROPERTY = 4
+const VALUE = 5
+const PARENT_SELECTOR = 6
+const SPACE = 7
+const CONDITION = 8
+```
+
+#### RULE_START
+
+Denotes the beginning of a rule, has a rule type.
+
+`[RULE_START, RULE_TYPE]`
+
+All rule types from https://wiki.csswg.org/spec/cssom-constants
+
+```
+1 STYLE_RULE CSSOM
+2 CHARSET_RULE CSSOM
+3 IMPORT_RULE CSSOM
+4 MEDIA_RULE CSSOM
+5 FONT_FACE_RULE CSSOM
+6 PAGE_RULE CSSOM
+7 KEYFRAMES_RULE css3-animations
+8 KEYFRAME_RULE css3-animations
+9 MARGIN_RULE CSSOM
+10 NAMESPACE_RULE CSSOM
+11 COUNTER_STYLE_RULE css3-lists
+12 SUPPORTS_RULE css3-conditional
+13 DOCUMENT_RULE css3-conditional
+14 FONT_FEATURE_VALUES_RULE css3-fonts
+15 VIEWPORT_RULE css-device-adapt
+16 REGION_STYLE_RULE proposed for css3-regions
+17 CUSTOM_MEDIA_RULE mediaqueries
+```
+
+#### RULE_END
+
+End of a rule.
+
+#### SELECTOR
+
+Denotes a selector or selector compound.
+
+Selector compound e.g. `.foo.bar` => `[SELECTOR, '.foo', '.bar']`
+
+Multiple classes e.g. `.foo .bar` => `[SELECTOR, '.foo'], [SELECTOR, '.bar']`
```

Comment:
This isn't right; the description down below is that you generate multiple `[SELECTOR ...]` tokens to represent comma-separated selectors, like `.foo, .bar`, which evaluate independently.

(The terms you want, btw, are "compound selector" for things like `.foo.bar`, "complex selector" when combinators are involved like `.foo .bar`, and "X selector list" when commas are involved, like `.foo, .bar`.)